x/go/calc/calc.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2023 Synnax Labs, Inc.
+//
+// Use of this software is governed by the Business Source License included in the file
+// licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with the Business Source
+// License, use of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt.
+
+package calc
+
+import (
+	"errors"
+	"go/ast"
+	"go/token"
+	"math"
+	"strconv"
+
+	"github.com/synnaxlabs/x/stack"
+)
+
+type Expression struct {
+	exp ast.Expr
+}
+
+type Resolver interface {
+	Resolve(string) (float64, error)
+}
+
+func findTokens(s string) ([]string, error) {
+	tokens := []string{}
+	currToken := ""
+	for i, c := range s {
+		if c == ' ' {
+			if currToken != "" {
+				tokens = append(tokens, currToken)
+				currToken = ""
+			}
+			continue
+		}
```

Comment:
Can we compress this into an `&&` statement and then use an `if else` block in the next clause instead of the `continue`?